### PR TITLE
Update numonly filter to output all Numeric values

### PR DESCRIPTION
**Pull request by CosmoPennypacker**
## Pull Request Checklist

**Is this in reference to an existing issue?**
Yes, [#102](https://github.com/sensu-plugins/sensu-plugins-http/issues/102)

#### General

- [ ] Update Changelog following the conventions laid out [here](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)

- [ ] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [ ] RuboCop passes

- [ ] Existing tests pass

#### New Plugins

- [ ] Tests

- [ ] Add the plugin to the README

- [ ] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

### CODE REVIEW

**Comment by majormoses** [comment]
Thanks for your contribution to Sensu plugins! Without people like you submitting PRs we couldn't run the project. I will review it shortly.

File: bin/metrics-http-json-deep.rb [hunk 1]
```diff
@@ -85,7 +85,7 @@ def deep_value(hash, scheme = '')
       if value.is_a?(Hash)
         deep_value(value, "#{scheme}.#{ekey}")
       else
-        output "#{scheme}.#{ekey}", value unless config[:numonly] && !value.is_a?(Integer)
+        output "#{scheme}.#{ekey}", value unless config[:numonly] && !value.is_a?(Numeric)
```
Comment (majormoses) [inline bin/metrics-http-json-deep.rb]: hmm, if we are gonna allow floats lets cap the number of decimals places, even better would be to have it make it a function param with a sane default.

Comment (CosmoPennypacker) [reply bin/metrics-http-json-deep.rb]: Could you please explain what exactly do you mean by a 'a function param'?
Does it mean changing the --number option from a flag to argument parameter (how is it defined?) and take the decimal places allowance? 
btw, I'm new to Ruby...

Comment (majormoses) [reply bin/metrics-http-json-deep.rb]: sorry I must have missed this but it looks like you figured out what I had in mind. Ping me once the CI issues have been fixed (or if you need help resolving them).

Comment (CosmoPennypacker) [reply bin/metrics-http-json-deep.rb]: I actually need help with resolving them, as the log doesn't supply useful information for understanding what's going on in the build. Thanks!

Comment (majormoses) [reply bin/metrics-http-json-deep.rb]: So taking a look [here](https://travis-ci.org/sensu-plugins/sensu-plugins-http/jobs/413943718#L1814-L1815) it looks like its not related to your changes as you have not updated the dependencies. I will try to find some time to dig through the dependencies and see where `net-telnet` is coming from and lock the right dependency to ensure compatibility for ruby versions `< 2.3` and then will make a major release removing support for ruby `< 2.3` as its EOL and per [our policy](https://github.com/sensu/sensu-docs/blob/master/content/plugins/1.0/faq.md#what-is-the-policy-on-supporting-end-of-lifeeol-ruby-versions) we attempt to support non EOL versions of ruby.

I will figure out how to get this over the line.

**Comment by majormoses** [comment]
https://github.com/sensu-plugins/sensu-plugins-http/pull/103#discussion_r209093664

**Comment by majormoses** [comment]
I also noticed that upon closer inspection the default was `10` I think thats too high of a default and will change it to `4` unless there is some reason why `10` makes sense.

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-http/versions/3.0.0

**Pull request merged**